Documentation/API.md

```diff
@@ -198,41 +239,123 @@ invisible to the user.  The property will be serialized if it has a non-default
 value.
 
 **Proto2 `optional` fields** generate properties of the corresponding
-type above.  It also generates `has` and `clear` methods that can be used to
+type above.
+It also generates `has` and `clear` methods that can be used to
 test whether the field has a value or to reset it to it's default.
-
-**Proto2 `optional` fields with default values** If the proto file
-specified a default value, that value will appear when the field has
-not been set or after it has been cleared.
+If a default value was specified in the proto file, the field will be
+initialized to that value, and will be reset to that value when
+you invoke the `clear` method.
+If no default value was specified, the default value is the same
+as for proto3 singular fields above.
 
 **Proto2 `required` fields** Required fields behave the same as
 optional fields, except that serialization or deserialization may
 fail if the field is not provided.
 
+To illustrate the handling of proto2 fields, consider the following
+short example:
+```protobuf
+syntax = "proto2";
+message ExampleProto2 {
+   int32 item_count = 1 [default = 12];
+   string item_label = 2;
+}
+```
+
+This will generate the following field structure in the Swift code:
+```swift
+public struct ExampleProto2 {
+    public var itemCount: Int32 = 12
+    public var hasItemCount: Bool
+    public mutating func clearItemCount()
+
+    public var itemLabel: String = "";
+    public var hasItemLabel: Bool
+    public mutating func clearItemLabel()
+}
+```
+
 ### Compound types
 
+**Singular message fields** generate simple properties of the corresponding
+Swift struct type.
+The fields are initialized with default instances of the struct.
+(This initialization is usually done lazily the first time you read such
+a field.)
+Message fields generate `has` and `clear` methods as above for both proto2
+and proto3.
+
+**Proto2 groups** act exactly like messages in all respects, except that they cannot be
+directly serialized on their own.
```

Comment:
Actually they can, they are just just messages.  The difference is the binary format is different when it is serialized as a field.